sdk/include/fail-simulator-on-error.h

```diff
@@ -59,7 +59,7 @@ compartment_error_handler(ErrorState *frame, size_t mcause, size_t mtval)
 		if (registerNumber == CHERI::RegisterNumber::CRA &&
 		    returnCapability.address() == 0 &&
 		    exceptionCode == CHERI::CauseCode::TagViolation &&
-		    stackCapability.top() == stackCapability.address())
+		    (stackCapability.top() - 16) == stackCapability.address())
```

Comment:
```suggestion
		    (stackCapability.top() - STACK_RESERVED_SPACE) == stackCapability.address())
```
or at least a comment explaining the magic 16.

Reply:
Thanks for the comment. I guess you don't want to include trusted stack headers here but by not using the symbolic name someone could change the value of `STACK_ENTRY_RESERVED_SPACE` and miss this. Consequences are not too grave, but you could at least use the symbol name in the comment so that it shows up in search?

Reply:
CI fails if you get this wrong.